MSAL/test/unit/native_auth/public/MSALNativeAuthUserAccountResultTests.swift

```diff
@@ -92,16 +94,102 @@ class MSALNativeAuthUserAccountResultTests: XCTestCase {
+        silentTokenProviderFactoryMock = MSALNativeAuthSilentTokenProviderFactoryMock()
 
         sut = MSALNativeAuthUserAccountResult(
             account: account!,
             rawIdToken: rawIdToken,
             configuration: MSALNativeAuthConfigStubs.configuration,
-            cacheAccessor: cacheAccessorMock
+            cacheAccessor: cacheAccessorMock,
+            silentTokenProviderFactory: silentTokenProviderFactoryMock
         )
         try super.setUpWithError()
     }
 
+    // MARK: - get access token tests
+
+    func test_getAcessToken_succesfullyReturnsAccessToken() async {
+        let accessToken = MSIDAccessToken()
+        accessToken.accessToken = "accessToken"
+        accessToken.scopes = ["scope1", "scope2"]
+        let contextCorrelationId = UUID()
+        let homeAccountId = MSALAccountId(accountIdentifier: "fedcba98-7654-3210-0000-000000000000.00000000-0000-1234-5678-90abcdefffff", objectId: "", tenantId: "https://contoso.com/tfp/tenantName")
+        let idToken = "newIdToken"
+        let account = MSALAccount(username: "1234567890", homeAccountId: homeAccountId, environment: "contoso.com", tenantProfiles: [])!
+        let silentTokenResult = MSALNativeAuthSilentTokenResult(accessTokenResult: MSALNativeAuthTokenResult(accessToken: accessToken.accessToken,
+                                                                                                             scopes: accessToken.scopes?.array as? [String] ?? [],
+                                                                                                             expiresOn: nil),
+                                                                rawIdToken: idToken,
+                                                                account: account,
+                                                                correlationId: contextCorrelationId)
+        silentTokenProviderFactoryMock.silentTokenProvider.result = silentTokenResult
+
+        let delegateExp = expectation(description: "delegateDispatcher delegate exp")
+        let expectedResult = MSALNativeAuthTokenResult(accessToken: accessToken.accessToken,
+                                                       scopes: accessToken.scopes?.array as? [String] ?? [],
+                                                       expiresOn: accessToken.expiresOn)
+
+        let delegate = CredentialsDelegateSpy(expectation: delegateExp, expectedResult: expectedResult)
+        delegate.expectedAccessToken = accessToken.accessToken
+        delegate.expectedScopes = accessToken.scopes?.array as? [String] ?? []
+        sut.getAccessToken(delegate: delegate)
+
+        await fulfillment(of: [delegateExp])
+
+        XCTAssertEqual(delegate.expectedResult, expectedResult)
+        XCTAssertEqual(sut.idToken, idToken)
+        XCTAssertEqual(sut.account, account)
+    }
+
+    func test_getAcessTokenScopesAndForceRefresh_succesfullyReturnsNewAccessToken() async {
```

Comment:
Passing either true or false in forceRefresh has no real impact in the test, since we are mocking the provider. Maybe we can remove one of these two tests.

Reply:
Indeed, the force refresh does nothing, but the scopes calls the other function, as there are 2 in the `MSALNativeAuthUserAccountResult` `getAccessToken(forceRefresh` and `getAccessToken(scopes`.
Each test calls a different function. I can remove the force refresh, but I would say to keep both function calls.